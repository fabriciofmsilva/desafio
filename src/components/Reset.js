import React from 'react';
import { connect } from 'react-redux';

import { reset } from '../actions/game';

export class Reset extends React.Component {

  reset = () => {
    this.props.reset();
  };

  render() {
    return (
      <div>
        <button
          disabled={!this.props.gameover}
          onClick={this.reset}
        >
          Reiniciar
        </button>
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  gameover: state.game.over
});

const mapDispatchToProps = (dispatch, props) => ({
  reset: (playerId) => dispatch(reset(playerId))
});

export default connect(mapStateToProps, mapDispatchToProps)(Reset);
