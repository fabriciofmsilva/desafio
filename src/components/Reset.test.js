import React from 'react';
import { shallow } from 'enzyme';

import { Reset } from './Reset';

let wrapper;
let reset;

beforeEach(() => {
  reset = jest.fn();
  wrapper = shallow(<Reset reset={reset} />);
});

test('should render Reset correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onClick', () => {
  wrapper.find('button').simulate('click');
  expect(reset).toHaveBeenCalled();
});
