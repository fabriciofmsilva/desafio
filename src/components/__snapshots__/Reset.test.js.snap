// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Reset correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reset
    reset={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="button"
        disabled={true}
        onClick={[Function]}
>
        Reiniciar
</button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Reiniciar",
        "className": "button",
        "disabled": true,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Reiniciar",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="button"
          disabled={true}
          onClick={[Function]}
>
          Reiniciar
</button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Reiniciar",
          "className": "button",
          "disabled": true,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reiniciar",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
